### DISCUSSION TITLE
Combined tree sequence of multiple chromosomes

### DISCUSSION THREAD

**Discussion by benekirsch** [tskit-dev/tskit#2517] [Q&A] [answered]
I have inferred and dated tree sequences for roughly 2000 haploid individuals. I obtained those via running tsinfer and tsdate on VCFs for each of the 22 human autosomes. 

I guess my question might be somewhat related to https://github.com/tskit-dev/tskit/discussions/2421 , but I might be way off. Is there a way to generate one tree sequence from my 22 tree sequences.

I am basically looking for something like " all = some_merge_function(ts1:ts22) ". Is that something achievable?

Best
Bene

**Comment by jeromekelleher** [comment]
It's technically do-able @benekirsch, but I'm not sure it would mean very much. There wouldn't be any nodes shared across the chromosomes so there wouldn't be any benefit of shared information, and it would mean that all chromosomes would have to be on a single long coordinate space.

Why do you want to merge them?

**Reply by benekirsch** [reply]
Thanks for the quick reply! Well, I guess I have a conceptual problem then because I would definitely want to have shared information. Am I guessing right, that I would have to infer on one 'long' chromosome to achieve that?

**Reply by jeromekelleher** [reply]
Yes, exactly. You'd have to have cross-chromosome phasing in order to make this in any way feasible, which I'm not sure anyone does.

@hyanwong - I think you've thought about this?

**Reply by apragsdale** [reply]
> You'd have to have cross-chromosome phasing in order to make this in any way feasible, which I'm not sure anyone does.

This is probably not relevant to the exact problem and data you have here, but this reminded me of a recent preprint on determining parent of origin for whole-autosome phasing (without needing parent-offspring trio data, but requiring methylation data), which would give you cross-chromosome phasing. https://www.biorxiv.org/content/10.1101/2022.05.24.493320v1.full

**Reply by hyanwong** [reply]
Yes, I think it would be a really nice thing to do for simulated data, and we have discussed how to encode it in tskit (see e.g. https://github.com/tskit-dev/tskit/issues/176#issuecomment-1076391522). For inferred data, as Jerome says, it's more difficult, unless you are (a) sequencing haploids (e.g. gametes or male bees) or (b) you have parent-offspring trios to phase across chromosomes or (c) you can somehow infer the chromosome phasing from two tree sequences containing the sample samples.

(c) is particularly interesting, but we haven't investigated it yet. Clearly it would be possible for e.g. someone like myself who has one European and one southeast Asian parent. I suspect you might be able to do something similar for most people, by looking for GNN patterns that distinguish their two (say) chromosome 1's, then map that onto chromosome 2, etc. That sort of approach would need a lot of testing first (are you volunteering?)

Another approach would be to figure out how to carry out tsinfer on unphased data. We have plans for getting the sample matching  phase working in this way, but we have no real idea how to do the ancestor generation step on unphased data. Anyway, I suspect that most ancestors won't consist of multiple chromosomes (although some recent ones might). You could look at this by carrying out some multi-chromosome simulations (see https://tskit.dev/msprime/docs/latest/ancestry.html#multiple-chromosomes). At least that way you'll be able to see the sort of pattern you expect for ancestors in the distant past.

**Comment by hyanwong** [comment, marked as answer]
Just to point you to the new `ts.concatenate()` function, which will do what you want but also not share nodes between chromosomes (apart from the sample nodes). Re cross-chromosomal phasing, there's some nice info in asides in this recent paper: https://www.nature.com/articles/s41586-025-09357-5

**Comment by hyanwong** [comment]
Hi @benekirsch  - I am just getting around to thinking about this (maybe too late for you!). See https://github.com/hyanwong/tsdict (there is also a slack channel for discussion, if you want)
